src/java.base/share/classes/java/lang/Process.java

```diff
@@ -161,6 +169,193 @@ public Process() {}
+     * @throws NullPointerException if the {@code charset} is {@code null}
+     * @since 17
+     */
+    public BufferedReader inputReader(Charset charset) {
+        return new BufferedReader(new InputStreamReader(getInputStream(), charset));
+    }
+
+    /**
+     * Returns a {@link BufferedReader BufferedReader} connected to the standard
+     * error of the process. The {@link Charset} for the native encoding is used
+     * to read characters, lines, or stream lines from standard error.
+     *
+     * <p>This method delegates to {@link #errorReader(Charset)} using the
+     * {@link Charset} named by the {@systemProperty native.encoding}
+     * system property or the {@link Charset#defaultCharset()} if the
+     * {@code native.encoding} is not supported.
+     *
+     * @return a {@link BufferedReader BufferedReader} using the
+     *          {@code native.encoding} if supported, otherwise, the
+     *          {@link Charset#defaultCharset()}
+     * @since 17
+     */
+    public BufferedReader errorReader() {
+        return errorReader(CharsetHolder.nativeCharset());
+    }
+
+    /**
+     * Returns a {@link BufferedReader BufferedReader} connected to the
+     * standard error of this process using a Charset.
+     * The {@code BufferedReader} can be used to read characters, lines,
+     * or stream lines of the standard error.
+     *
+     * <p>Characters are read by an InputStreamReader that reads and decodes bytes
+     * from this process {@link #getErrorStream()}. Bytes are decoded to characters
+     * using the {@code charset}; malformed-input and unmappable-character
+     * sequences are replaced with the charset's default replacement.
+     * The BufferedReader reads and buffers characters from the InputStreamReader.
+     *
+     * <p>If the standard error of the process has been redirected using
+     * {@link ProcessBuilder#redirectError(Redirect) ProcessBuilder.redirectError} or
+     * {@link ProcessBuilder#redirectErrorStream(boolean) ProcessBuilder.redirectErrorStream}
+     * then the {@code InputStreamReader} will be reading from a
+     * <a href="ProcessBuilder.html#redirect-output">null input stream</a>.
+     *
+     * @apiNote
+     * Using both {@link #getErrorStream} and {@link #errorReader} has
+     * unpredictable behavior since the buffered reader reads ahead from the
+     * error stream.
+     *
+     * @implNote
+     * This is equivalent to:
+     * {@code
+     *     return new BufferedReader(new InputStreamReader(getErrorStream(), charset));
+     * }
+     *
+     * @param charset the {@code Charset} used to decode bytes to characters, not null
+     * @return a BufferedReader for the standard error of the process using the {@code charset}
+     * @throws NullPointerException if the {@code charset} is {@code null}
+     * @since 17
+     */
+    public BufferedReader errorReader(Charset charset) {
+        return new BufferedReader(new InputStreamReader(getErrorStream(), charset));
```

Comment:
Same remark here: I'd suggest calling Objects.requireNonNull(charset) first thing in this method instead of relying on the InputStreamReader constructor to throw the NPE.